Demos/api/Orchestration/Autonomous.cpp

```diff
@@ -0,0 +1,77 @@
+// SPDX-FileCopyrightText: 2024 Vector Informatik GmbH
+//
+// SPDX-License-Identifier: MIT
+
+#include <iostream>
+#include <thread>
+
+#include "silkit/SilKit.hpp"
+
+using namespace std::chrono_literals;
+
+int main(int argc, char** argv)
+{
+    if (argc != 2)
+    {
+        std::cerr << "Wrong number of arguments! Start demo with: " << argv[0] << " <ParticipantName>" << std::endl;
+        return -1;
+    }
+    std::string participantName(argv[1]);
+
+    try
+    {
+        // Setup participant, lifecycle, time synchronization and logging.
+        const std::string registryUri = "silkit://localhost:8500";
+        const std::string configString = R"({"Logging":{"Sinks":[{"Type":"Stdout","Level":"Info"}]}})";
+        auto participantConfiguration = SilKit::Config::ParticipantConfigurationFromString(configString);
+
+        auto participant = SilKit::CreateParticipant(participantConfiguration, participantName, registryUri);
+        auto logger = participant->GetLogger();
+
+        // Create an autonomous lifecycle 
+        auto* lifecycleService =
+            participant->CreateLifecycleService({SilKit::Services::Orchestration::OperationMode::Autonomous});
+
+
+         // Future / promise to control entrance of the main loop in the worker thread
+        std::promise<void> startWorkPromise;
+        std::future<void> startWorkFuture;
+        startWorkFuture = startWorkPromise.get_future();
+
+        // The worker thread is 'unleashed' in the starting handler...
+        lifecycleService->SetStartingHandler(
+            [&startWorkPromise]() { startWorkPromise.set_value(); });
+
+        // Start the worker thread and wait for the go from the starting handler.
+        auto workerThread = std::thread([&startWorkFuture, lifecycleService, logger]() {
+            startWorkFuture.wait();
```

Comment:
The thread will be 'automatically' terminated (via exception). Probably also wrap the whole thing in a `try { ... } catch { ... }`.

```suggestion
            startWorkFuture.get();

```

Reply:
Problem with `.wait()` is that it just returns even if the promise is destroyed or has an exception set.